src/query/functions/tests/it/scalars/testdata/array.txt

```diff
@@ -1756,7 +1756,7 @@ error:
   --> SQL:1:12
   |
 1 | array_sort(['x', 0, 1.2, 3.4, 5.6, 7.8])
-  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ unable to cast type `String` to type `Decimal(2, 1)`
+  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ fail to decode string as decimal: Code: 1006, displayText = bad decimal literal: unexpected char. while evaluating function `to_decimal("x")`
```

Comment:
Please also add some unit tests for decimal functions.

Reply:
this pr is about to_decimal(string), 

since we do not allow auto cast from string to decimal, we only need to test manual cast ?  there is a simple one in this pr.  since the core function invoked is  the same as parsing a decimal expr, I did not repeate these tests.

which kind of additional tests do you suggest?

Reply:
like this `test_to_string` function.
https://github.com/datafuselabs/databend/blob/main/src/query/functions/tests/it/scalars/arithmetic.rs#L108

Reply:
this pr should not change this file I think,  if string auto cast to decimal is not allowed.

Reply:
but  string auto cast to  int/float, now, you can try:

```
mysql> select '1' + 11;
+------------+
| ('1' + 11) |
+------------+
|         12 |
+------------+
1 row in set (0.04 sec)
Read 1 rows, 1.00 B in 0.004 sec., 247.45 rows/sec., 247.45 B/sec.

mysql> select '1' + 11.1::float32;
+-----------------------+
| ('1' + 11.1::float32) |
+-----------------------+
|    12.100000381469727 |
+-----------------------+
```

Reply:
```
fn register_auto_cast_rules(registry: &mut FunctionRegistry) {
    registry.register_default_cast_rules(GENERAL_CAST_RULES.iter().cloned());
    registry.register_default_cast_rules(CAST_FROM_STRING_RULES.iter().cloned());
```

Reply:
but CAST_FROM_STRING_RULES does not allow string to decimal of float.

Reply:
> this pr should not change this file I think, if string auto cast to decimal is not allowed.

the super type of  ```['x', 0, 1.2, 3.4, 5.6, 7.8]```  is decimal.

since  string -> unit8 -> decimal is auto

when executing,  try string-> decimal, and got the error

```
  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ fail to decode string as decimal: Code: 1006, displayText = bad decimal literal: unexpected char. while evaluating function `to_decimal("x")`
```

Reply:
Oh, I see that now.